erc20/ds-token/ds-token-erc20-spec.ini

```diff
@@ -0,0 +1,265 @@
+
+[transfer-failure-2-a]
++requires:
+    andBool VALUE >Int BAL_FROM
+
+[transfer-failure-2-b]
++requires:
+    andBool STOPPED ==Int 1 <<Int (20 *Int 8)
+
+
+[transferFrom]
+callData: #abiCallData("transferFrom", #address(FROM_ID), #address(TO_ID), #uint256(VALUE))
+gas: {GASCAP} => _
+refund: _ => _
+storage:
+    #hashedLocation("Solidity", {_STOPPED}, .IntList) |-> STOPPED
+requires:
+    andBool 0 <=Int FROM_ID   andBool FROM_ID   <Int (2 ^Int 160)
+    andBool 0 <=Int TO_ID     andBool TO_ID     <Int (2 ^Int 160)
+    andBool 0 <=Int VALUE     andBool VALUE     <Int (2 ^Int 256)
+    andBool 0 <=Int BAL_FROM  andBool BAL_FROM  <Int (2 ^Int 256)
+    andBool 0 <=Int BAL_TO    andBool BAL_TO    <Int (2 ^Int 256)
+    andBool 0 <=Int ALLOW     andBool ALLOW     <Int (2 ^Int 256)
+    andBool 0 <=Int STOPPED    andBool STOPPED    <Int (2 ^Int 256)
+
+[transferFrom-success]
+k: #execute => (RETURN RET_ADDR:Int 32 ~> _)
+localMem: .Map => ( .Map[ RET_ADDR := #asByteStackInWidth(1, 32) ] _:Map )
+log: _ => _
+
+[transferFrom-success-1]
++storage:
+    #hashedLocation({COMPILER}, {_BALANCES},   FROM_ID)           |-> (BAL_FROM => BAL_FROM -Int VALUE)
+    #hashedLocation({COMPILER}, {_BALANCES},   TO_ID)             |-> (BAL_TO   => BAL_TO   +Int VALUE)
+    #hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID) |-> (ALLOW    => ALLOW    -Int VALUE)
+    _:Map
++requires:
+    andBool FROM_ID =/=Int TO_ID
+    andBool VALUE <=Int BAL_FROM
+    andBool BAL_TO +Int VALUE <Int (2 ^Int 256)
+    andBool VALUE <=Int ALLOW
+    andBool STOPPED ==Int 0
+
+[transferFrom-success-2]
++storage:
+    #hashedLocation({COMPILER}, {_BALANCES},   FROM_ID)           |-> BAL_FROM
+    #hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID) |-> (ALLOW => ALLOW -Int VALUE)
+    _:Map
++requires:
+    andBool FROM_ID ==Int TO_ID
+    andBool VALUE <=Int BAL_FROM
+    andBool VALUE <=Int ALLOW
+    andBool STOPPED ==Int 0
+
+[transferFrom-failure]
+k: #execute => #exception
+localMem: .Map => _:Map
+log: _ => _
+
+[transferFrom-failure-1]
++storage:
+    #hashedLocation({COMPILER}, {_BALANCES},   FROM_ID)           |-> (BAL_FROM => _)  // BAL_FROM
+    #hashedLocation({COMPILER}, {_BALANCES},   TO_ID)             |-> (BAL_TO   => _)  // BAL_TO
+    #hashedLocation({COMPILER}, {_ALLOWANCES}, FROM_ID CALLER_ID) |-> ALLOW
+    _:Map
++requires:
+    andBool FROM_ID =/=Int TO_ID
+
+[transferFrom-failure-1-a]
++requires:
+    andBool VALUE >Int BAL_FROM
+
+
+[transferFrom-failure-1-b]
+storage:
+    #hashedLocation("Solidity", 4, .IntList) |-> STOPPED
+    #hashedLocation("Solidity", 1,   FROM_ID)           |-> (BAL_FROM => _)  // BAL_FROM
+    #hashedLocation("Solidity", 1,   TO_ID)             |-> BAL_TO           // BAL_TO
+    #hashedLocation("Solidity", 2, FROM_ID CALLER_ID) |-> (ALLOW => _)
+    _:Map
++requires:
+    andBool VALUE <=Int BAL_FROM
+    andBool VALUE <=Int ALLOW
+    andBool BAL_TO +Int VALUE >=Int (2 ^Int 256)
```

Comment:
No. The lines are correct because this is the overflow case. So ALLOW and BAL_FROM are changed and BAL_TO don't. And the 2 additional lines from requires are added because k was testing wrong paths